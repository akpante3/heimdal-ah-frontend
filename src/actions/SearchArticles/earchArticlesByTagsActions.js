import instance from '../../config/http';
import {
  ACTIONS,
} from '../actionTypes';
import ActionResponse from '../actionResponse';

const searchArticleByTags = query => dispatch => instance.get(`/articles_search/tag?tag=${query}&offset=0&size=7`).then((response) => {
  if (response.status === 200) {
    dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_TAGS_SUCCESS,
      response.data.articles.rows));
  } else if (response.status === 404) {
    dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_TAGS_FAILED, 'Sorry, no articles found matching your search parameters'));
  }
}).catch((error) => {
  dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_TAGS_ERROR, 'Server error', error));
});

export default searchArticleByTags;
