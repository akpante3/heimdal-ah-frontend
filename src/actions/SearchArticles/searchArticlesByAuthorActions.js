import instance from '../../config/http';
import {
  ACTIONS,
} from '../actionTypes';
import ActionResponse from '../actionResponse';

let http;
const searchArticleByAuthor = (query, field, size) => (dispatch) => {
  if (field !== null && size !== null) {
    http = instance.get(`/articles_search/author?author=${query}&offset=${field}&size=${size}`);
  } else {
    http = instance.get(`/articles_search/author?author=${query}`);
  }

  http.then((response) => {
    if (response.status === 200) {
      dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_AUTHOR_SUCCESS,
        response.data.articles.rows));
    } else if (response.status === 404) {
      dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_AUTHOR_FAILED, 'Sorry, no articles found matching your search parameters'));
    }
  }).catch((error) => {
    dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_AUTHOR_ERROR, 'Server error', error));
  });
};

export default searchArticleByAuthor;
