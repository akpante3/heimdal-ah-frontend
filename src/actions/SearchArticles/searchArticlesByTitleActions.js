import instance from '../../config/http';
import {
  ACTIONS,
} from '../actionTypes';
import ActionResponse from '../actionResponse';

const searchArticleByTitle = query => dispatch => instance.get(`/articles_search/title?title=${query}&offset=0&size=7`).then((response) => {
  if (response.status === 200) {
    dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_TITLE_SUCCESS, response.data.articles.rows));
    // dispatch(ActionResponse(ACTIONS.RESET_SEARCH_ARTICLES_STATE));
  } else if (response.status === 404) {
    dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_TITLE_FAILED, 'Sorry, no articles found matching your search parameters'));
    // dispatch(ActionResponse(ACTIONS.RESET_SEARCH_ARTICLES_STATE));
  }
}).catch((error) => {
  dispatch(ActionResponse(ACTIONS.SEARCH_ARTICLES_BY_TITLE_ERROR, 'Server error', error));
//   dispatch(ActionResponse(ACTIONS.RESET_SEARCH_ARTICLES_STATE));
});

export default searchArticleByTitle;
