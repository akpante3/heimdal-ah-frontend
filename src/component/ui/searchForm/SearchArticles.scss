@import '../../../styles/variables.scss';

.search-articles {
    table {
        background-color: $white-color;
        border-collapse: collapse;
        box-shadow: 2px 2px 6px rgb(136, 134, 134);
        color: $search-articles__tr--bg-color;
        font-family: 'Roboto, sans-serif';
        font-size: 17px;
        position: absolute;
        right: 4.5%;
        top: 3.6em;
        width: 30%;
        z-index: 2;
    }
    
    td, th {
        padding: 10px 10px 10px 30px;
    }
    
    a {
        color: $heimdal-textfield-background;
    }
    
    a:hover {
        opacity: 0.5;
        text-decoration: none;
    }
    
    // tr:nth-child(even) {
    //     color: $white-color;
    //     background-color: $search-articles__tr--bg-color;
    // }
    
    // tr:nth-child(even) a {
    //     color: $white-color;
    // }
}

.search-articles-full-body {
   &.header-formm {
    margin: 4% 30% 2% 30%;
   }

    table {
        background-color: $white-color;
        border-collapse: collapse;
        box-shadow: 2px 2px 6px rgb(136, 134, 134);
        color: $search-articles__tr--bg-color;
        font-family: 'Roboto, sans-serif';
        font-size: 20px;
        left: 4%;
        margin-bottom: 10%;
        position: relative;
        right: 2%;
        width: 92%;
    }
    
    td, th {
        padding: 3px 3px 3px 20px;
        border-bottom: 1px solid rgb(173, 171, 171);
    }

    th {
        background-color: rgb(122, 187, 212);;
        color: $black;
    }

    td:hover {
        background-color: rgba(122, 187, 212, 0.2);
    }
    
    a {
        color: $black;
    }
    
    a:hover {
        opacity: 0.5;
        color: blueviolet;
        text-decoration: none;
    }
    
    // tr:nth-child(even) {
    //     color: $white-color;
    //     background-color: plum;
    // }
    
    // tr:nth-child(even) a {
    //     color: $white-color;
    // }
}