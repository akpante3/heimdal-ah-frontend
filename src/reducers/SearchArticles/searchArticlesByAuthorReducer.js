import {
  ACTIONS,
} from '../../actions/actionTypes';

const initialState = {
  payload: '',
  status: '',
};

export default (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS.SEARCH_ARTICLES_BY_AUTHOR_FAILED:
      return ({
        ...state,
        payload: action.payload,
        status: 'FAILED',
      });

    case ACTIONS.SEARCH_ARTICLES_BY_AUTHOR_SUCCESS:
      return ({
        ...state,
        payload: action.payload,
        status: 'SUCCESS',
      });

    case ACTIONS.SEARCH_ARTICLES_BY_AUTHOR_ERROR:
      return ({
        ...state,
        payload: action.payload,
        status: 'ERROR',
      });

    default:
      return (state);
  }
};
