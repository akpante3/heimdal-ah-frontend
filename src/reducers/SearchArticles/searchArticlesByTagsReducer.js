import {
  ACTIONS,
} from '../../actions/actionTypes';

const initialState = {
  payload: '',
  status: '',
};

export default (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS.SEARCH_ARTICLES_BY_TAGS_FAILED:
      return ({
        ...state,
        payload: action.payload,
        status: 'FAILED',
      });

    case ACTIONS.SEARCH_ARTICLES_BY_TAGS_SUCCESS:
      return ({
        ...state,
        payload: action.payload,
        status: 'SUCCESS',
      });

    case ACTIONS.SEARCH_ARTICLES_BY_TAGS_ERROR:
      return ({
        ...state,
        payload: action.payload,
        status: 'ERROR',
      });

    default:
      return (state);
  }
};
