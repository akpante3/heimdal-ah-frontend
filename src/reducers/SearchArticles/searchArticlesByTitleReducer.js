import {
  ACTIONS,
} from '../../actions/actionTypes';

const initialState = {
  payload: '',
  status: '',
};

export default (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS.RESET_SEARCH_ARTICLES_BY_TITLE_STATE:
      return ({
        ...state,
        payload: '',
        status: '',
      });

    case ACTIONS.SEARCH_ARTICLES_BY_TITLE_FAILED:
      return ({
        ...state,
        payload: action.payload,
        status: 'FAILED',
      });

    case ACTIONS.SEARCH_ARTICLES_BY_TITLE_SUCCESS:
      return ({
        ...state,
        payload: action.payload,
        status: 'SUCCESS',
      });

    case ACTIONS.SEARCH_ARTICLES_BY_TITLE_ERROR:
      return ({
        ...state,
        payload: action.payload,
        status: 'ERROR',
      });

    default:
      return (state);
  }
};
