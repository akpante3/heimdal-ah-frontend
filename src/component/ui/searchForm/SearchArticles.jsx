/* eslint-disable no-nested-ternary */
import React, { Fragment, Component } from 'react';
import { connect } from 'react-redux';
import { FormControl } from 'react-bootstrap';
import searchArticleByAuthor from '../../../actions/SearchArticles/searchArticlesByAuthorActions';
import searchArticleByTitle from '../../../actions/SearchArticles/searchArticlesByTitleActions';
import searchArticleByTags from '../../../actions/SearchArticles/earchArticlesByTagsActions';
import './SearchArticles.scss';
// import Alert from '../Alert/Alert';

/**
* @description - Helps a user rate an article
* @param {props} status - the status returned from dispatching articleRating action
* @param {props} star - the star rating given by a user to the article
* @param {props} getratingstatus - the status returned from dispatching getArticleRating action
* @param {props} ratings - average rating for the article
* @returns {component} Component
*/
class SearchArticles extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayInput: false,
      query: null,
    };
  }

  /**
* @description - Handles the initial component mounting and calls getArticleRating action
*/
  // componentDidMount() {
  //   const articleId = localStorage.getItem('articleId');
  //   this.props.actions(getArticleRating(Number(articleId)));
  // }

  /**
* @description - Handles the submission of the rating by calling articleRating action
* @param {number} stars - the star rating of the article, supplied by user
*/
  // onHandleSubmit(stars) {
  //   // Might change based on where the article id is  gotten from
  //   const articleId = localStorage.getItem('articleId');
  //   // eslint-disable-next-line react/destructuring-assignment
  //   this.props.actions(articleRating(stars, articleId));
  // }

  onHandleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
    setTimeout(() => {
      this.props.actions(searchArticleByAuthor(this.state.query));
      this.props.actions(searchArticleByTitle(this.state.query));
      this.props.actions(searchArticleByTags(this.state.query));
    }, 1);
    // this.props.actions(getArticlesByAuthoor(query));
    // this.props.actions(getArticlesByTags(query));
  }

  /**
* @description - Handles the rendering of the component
* @returns {component} the rendered component
*/
  render() {
    const { displayInput } = this.state;
    return (
      <Fragment>
        <form className="header-form form-inline my-2 my-lg-0">
          <div className={displayInput ? '' : 'header-input'}>
            <FormControl
              className='form-control mr-sm-2'
              placeholder='Search....'
              name='query'
              value={this.state.query}
              onChange={e => this.onHandleChange(e)}
            />
          </div>
          <div
            role="searchbox"
            onKeyPress={() => this.setState({ displayInput: !displayInput })}
            onClick={() => this.setState({ displayInput: !displayInput })}
            className='search_icon'
            tabIndex="-1"
          >
            <i className="fas fa-search" />
          </div>
        </form>
        {this.props.articlesByAuthorStatus === 'SUCCESS' ? (
          <div className='search-articles'>
            <br />
            <table>
              <tbody>
                <tr>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Tags
                  </th>
                </tr>
                {this.props.articlesByAuthor.map(article => (
                  <tr className='search-articles' key={article.id}>
                    <td>
                      <a href={`https://heimdal-frontend.herokuapp.com/${article.user.profile.username}/${article.slug}`}>{article.title}</a>
                    </td>
                    <td>
                      {article.user.profile.username}
                    </td>
                  </tr>
                ))
                }
              </tbody>
            </table>
          </div>
        ) : this.props.articlesByTitleStatus === 'SUCCESS' ? (
          <div className='search-articles'>
            <br />
            <table>
              <tbody>
                <tr>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Tags
                  </th>
                </tr>
                {this.props.articlesByTitle.map(article => (
                  <tr className='search-articles' key={article.id}>
                    <td>
                      <a href={`https://heimdal-frontend.herokuapp.com/articles/${article.slug}`}>{article.title}</a>
                    </td>
                    <td>
                      Joshua
                    </td>
                  </tr>
                ))
                }
              </tbody>
            </table>
          </div>
        ) : this.props.articlesByTagsStatus === 'SUCCESS' ? (
          <div className='search-articles'>
            <br />
            <table>
              <tbody>
                <tr>
                  <th>
                    Title
                  </th>
                  <th>
                    Author
                  </th>
                  <th>
                    Tags
                  </th>
                </tr>
                {this.props.articlesByTags.map(article => (
                  <tr className='search-articles' key={article.id}>
                    <td>
                      <a href={`https://heimdal-frontend.herokuapp.com/articles/${article.article.slug}`}>{article.article.title}</a>
                    </td>
                    <td>
                      Joshua
                    </td>
                    <td>
                      {article.tag.tagName}
                    </td>
                  </tr>
                ))
                }
              </tbody>
            </table>
          </div>
        ) : ''
        }
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  articlesByAuthorStatus: state.searcharticlesbyauthor.status,
  articlesByAuthor: state.searcharticlesbyauthor.payload,
  articlesByTitle: state.searcharticlesbytitle.payload,
  articlesByTitleStatus: state.searcharticlesbytitle.status,
  articlesByTagsStatus: state.searcharticlesbytags.status,
  articlesByTags: state.searcharticlesbytags.payload,
});

const mapDispatchToProps = dispatch => ({
  actions: action => dispatch(action),
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchArticles);
