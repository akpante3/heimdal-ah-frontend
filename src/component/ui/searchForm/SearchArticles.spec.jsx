/* eslint-disable react/no-children-prop */
/* eslint-disable no-undef */
import React from 'react';
import { SearchArticles } from './SearchArticles';

describe('Render <SearchArticles />', () => {
  const wrapper = shallow(<SearchArticles children={() => { }} actions={() => { }} articlesByAuthorStatus='SUCCESS' articlesByTitleStatus='SUCCESS' articlesByTagsStatus='SUCCESS' />);
  it('it should render  ArticleRating component', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

//   it('it should simulate change event', () => {
//     const spy = jest.spyOn(SearchArticles.prototype, 'onHandleChange');
//     console.log(wrapper.find('input'));
//     wrapper.find('input').at(0).simulate('change');
//     expect(spy).toHaveBeenCalled();
//   });
});
