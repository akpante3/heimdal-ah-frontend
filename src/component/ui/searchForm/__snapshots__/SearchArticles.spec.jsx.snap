// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <SearchArticles /> it should render  ArticleRating component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchArticles
    actions={[Function]}
    articlesByAuthorStatus="SUCCESS"
    articlesByTagsStatus="SUCCESS"
    articlesByTitleStatus="SUCCESS"
  >
    [Function]
  </SearchArticles>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
